backend/models/user.js

```diff
@@ -265,15 +266,15 @@ schema.statics.createUser = function(logger, username, password, customData, tok
 		}
 	
 
-		["firstName", "lastName", "email"].forEach(key => {
+		["firstName", "lastName", "email", "mailListOptOut"].forEach(key => {
 			if (customData && customData[key]){
```

Comment:
As discussed, let's do an early reject on customData and we can remove all the if(customData) checks after that.